packages/cbioportal-clinical-timeline/src/TrackHeader.tsx

```diff
@@ -19,57 +19,66 @@ export function getTrackLabel(track: TimelineTrackSpecification) {
 }
 
 const TrackHeader: React.FunctionComponent<ITrackHeaderProps> = function({
+    store,
     track,
     handleTrackHover,
     height,
     paddingLeft = 5,
 }) {
-    return (
+    const collapseCallback = useCallback(
+        () => store.toggleTrackCollapse(track.uid),
+        [track] // todo: would this work or do i have to use a normal props and dereference the prop here?
```

Comment:
notice @alisman

Reply:
not sure i understand the question.

Reply:
I want the callback to update when the prop `track` changes. So Im wondering whether it would work to list the dependency as `[track]` or whether I have to do the dereferencing in here like `props.track` or even `()=>props.track`.

Reply:
I think it depends on what track is.  My guess is that if track is still referentially equal, this will not work.